lib/matplotlib/tests/test_backend_tk.py

```diff
@@ -234,3 +234,37 @@ class Toolbar(NavigationToolbar2Tk):
         # Checking return code late so the stdout assertions happen first
         if proc.returncode:
             pytest.fail("Subprocess failed to test intended behavior")
+
+
+@pytest.mark.backend('TkAgg', skip_on_importerror=True)
+def test_toolbaar_zoompan():
+    script = """
+import matplotlib.pyplot as plt
+plt.rcParams['toolbar'] = 'toolmanager'
+ax = plt.gca()
+if ax.get_navigate_mode() is None:
+    print("success")
+ax.figure.canvas.manager.toolbar._button_click('zoom')
+if ax.get_navigate_mode() == "ZOOM":
+    print("success")
+ax.figure.canvas.manager.toolbar._button_click('pan')
```

Comment:
This just calls `self.trigger_tool`; can you not do that directly instead of using a private method? And then would it work with non-Tk backends? (Maybe not Agg if it doesn't have a toolbar, though.)

Reply:
Thank you. calling private method is not needed. 


>would it work with non-Tk backends? (Maybe not Agg if it doesn't have a toolbar, though.)

Can you also hint me how can I check if it will work for non-Tk backend?(maybe travis or azure tests in pipeline check this?) And in some configuration, is it possible to have no toolbar while running testcase?

Reply:
There's `test_interactive_zoom` in `test_backend_bases.py` that tests similar things without relying on a specific backend (it will use the default Agg, I believe.)

There's also `test_backend_tools.py`, but I don't think you could put anything in there as that doesn't attempt to create a window.

You could also test all backends by duplicating a test in `test_backends_interactive.py`.

Reply:
@QuLogic I tried to put this test case in `test_backend_bases.py`  with `trigger_tool`, it runs without error when I run it in python file. But when I run it using **pytest** it throws exception(`AttributeError: 'NoneType' object has no attribute 'trigger_tool'`):

```
def test_toolbar_zoompan():
    plt.rcParams['toolbar'] = 'toolmanager'
    ax = plt.gca()
    assert ax.get_navigate_mode() is None
    ax.figure.canvas.manager.toolbar.trigger_tool('zoom')
    assert ax.get_navigate_mode() == "ZOOM"
    ax.figure.canvas.manager.toolbar.trigger_tool('pan')
    assert ax.get_navigate_mode() == "PAN"
```